### ISSUE TITLE
better search in docs

### ISSUE THREAD

**Issue by ntabris** [dask/dask#9280]
My anecdotal experience is that the search bar in dask docs rarely gives me the page I actually want. It doesn't appear to do a good job on search term proximity or non-exact matches and maybe indexes too much on the API sections.

For example, if I search distributed docs for "worker plugin"

https://distributed.dask.org/en/stable/search.html?q=worker+plugin

it's quite challenging to see the result that I actually want.

Anyone know anything about improving / changing the search engine used for this functionality?

**Comment by scharlottej13** [comment]
Poking around on [search features](https://docs.readthedocs.io/en/stable/server-side-search.html#search-features) in the RTD docs I found a section on [configuring the search ranking](https://docs.readthedocs.io/en/stable/config-file/v2.html#search-ranking). Reading this it seems like we could make some pages more likely to show up than others, if that's something we wanted to explore.

@ntabris there's also this section on using [advanced search features](https://docs.readthedocs.io/en/stable/guides/advanced-search.html#using-advanced-search-features)-- not sure if you've already tried some of these techniques?

cc @jsignell @jacobtomlinson

**Comment by jrbourbeau** [comment]
> @ntabris there's also this section on using [advanced search features](https://docs.readthedocs.io/en/stable/guides/advanced-search.html#using-advanced-search-features)-- not sure if you've already tried some of these techniques?

FWIW I find the results of searching for `"worker plugin"` (with the quotes) to be more meaningful

**Comment by ntabris** [comment]
> FWIW I find the results of searching for "worker plugin" (with the quotes) to be more meaningful

Hm, those results seem even worse to me since the top hits are all API/docstrings and the page that actually explains worker plugins is much further down.

**Comment by jrbourbeau** [comment]
Fair point. I suppose it ultimately depends on what you're looking for. My previous comment https://github.com/dask/dask/issues/9280#issuecomment-1185808265 was because the top results were at least related to plugins as opposed to about the worker generally (but not related to the plugin system)
